test/test_xnan_functions.cpp

```diff
@@ -202,34 +202,73 @@ namespace xt
 
     TEST(xnanfunctions, result_type) {
         shape_type shape = {4, 3, 2};
+        xarray<short> ashort(shape);
+        xarray<unsigned short> aushort(shape);
         xarray<int> aint(shape);
+        xarray<unsigned int> auint(shape);
+        xarray<long long> along(shape);
+        xarray<unsigned long long> aulong(shape);
         xarray<float> afloat(shape);
         xarray<double> adouble(shape);
 
+#define CHECK_RESULT_TYPE_FOR_ALL(INPUT, RESULT_TYPE)                               \
+        CHECK_RESULT_TYPE(nansum(INPUT, {1, 2}), RESULT_TYPE);                      \
+        CHECK_RESULT_TYPE(nanmean(INPUT, {1, 2}), double);                          \
+        CHECK_RESULT_TYPE(nanvar(INPUT, {1, 2}), double);                           \
+        CHECK_RESULT_TYPE(nanstd(INPUT, {1, 2}), double);
+
+#define CHECK_TEMPLATED_RESULT_TYPE_FOR_ALL(INPUT, RESULT_TYPE)                     \
+        CHECK_RESULT_TYPE(nansum<RESULT_TYPE>(INPUT, {1, 2}), RESULT_TYPE)          \
+        CHECK_RESULT_TYPE(nanmean<RESULT_TYPE>(INPUT, {1, 2}), RESULT_TYPE)         \
+        CHECK_RESULT_TYPE(nanvar<RESULT_TYPE>(INPUT, {1, 2}), RESULT_TYPE)          \
+        CHECK_RESULT_TYPE(nanstd<RESULT_TYPE>(INPUT, {1, 2}), RESULT_TYPE)
+
+        /*********
+         * short *
+         *********/
+        CHECK_RESULT_TYPE_FOR_ALL(ashort, short);
+        CHECK_TEMPLATED_RESULT_TYPE_FOR_ALL(ashort, int);
```

Comment:
The result type is unintuitive to me. But I have little experience with `short int`.

Reply:
This arithmeric rule (promoting char and short to int before arithmetic operation) is inherited from C. See this [SO answer](https://stackoverflow.com/a/24372323/7241714) for more details